internal/ingest/ingest.go

```diff
@@ -147,14 +179,38 @@ func (i *legIngester) listenUpdates(ctx context.Context, s *sub) {
 	}
 }
 
+func (i *legIngester) listenSyncUpdates(ctx context.Context, p peer.ID,
+	watcher chan cid.Cid, cncl context.CancelFunc, out chan cid.Cid) {
+
+	defer func() {
+		cncl()
+		close(out)
+	}()
+
+	select {
+	case <-ctx.Done():
+		return
+	// Persist the latest sync
+	case c := <-watcher:
+		err := i.putLatestSync(p, c)
+		if err != nil {
+			log.Errorf("Error persisting latest sync: %w", err)
+		}
+		out <- c
+	}
+}
+
 // Unsubscribe to stop listening to advertisement from a specific provider.
 func (i *legIngester) Unsubscribe(ctx context.Context, p peer.ID) error {
 	i.sublk.Lock(p)
 	defer i.sublk.Unlock(p)
 	// Close subscriber
 	i.subs[p].ls.Close()
-	// Run cancel
-	i.subs[p].cncl()
+	// Check if we are subscribed
```

Comment:
Seems that there should be a check to make sure `i.subs[p]` returns a value.  Consider is `Unsubscribe` is called a second time.